lib/pure/strutils.nim

```diff
@@ -2860,6 +2860,48 @@ proc isNilOrWhitespace*(s: string): bool {.noSideEffect, procvar, rtl,
   ## Alias for isEmptyOrWhitespace
   result = isEmptyOrWhitespace(s)
 
+
+since (1, 1):
+  func toLowerAsciiInPlace*(s: var string) {.inline.} =
+    ## Converts string `s` into lower case.
+    ##
+    ## This works only for the letters ``A-Z``. See `unicode.toLower
+    ## <unicode.html#toLower,string>`_ for a version that works for any Unicode
+    ## character.
+    ##
+    ## See also:
+    ## * `normalize proc<#normalize,string>`_
+    runnableExamples:
+      var stringy = "ABCDEF"
+      toLowerAsciiInPlace(stringy)
+      doAssert stringy == "abcdef"
+      var strng = "NIM"
+      toLowerAsciiInPlace(strng)
+      doAssert strng == "nim"
+    for c in mitems(s):
+      if c in {'A'..'Z'}: c = chr(ord(c) + (ord('a') - ord('A')))
```

Comment:
Can you add `{.inline.}` to `to{Lower,Upper}Ascii(char)` then use them directly instead? Please don't reply "Done" to my suggestions then do it your way.

```suggestion
      c = toLowerAscii(c)
```